src/net_processing.cpp

```diff
@@ -3341,6 +3365,14 @@ void PeerLogicValidation::ConsiderEviction(CNode *pto, int64_t time_in_seconds)
             }
         }
     }
+
+    const auto& tx_download = state.m_tx_download;
+    if (tx_download.m_tx_in_flight.size() >= MAX_PEER_TX_IN_FLIGHT) {
+        if (!tx_download.m_tx_process_time.empty() && (tx_download.m_tx_process_time.begin()->first + GETDATA_TX_TIMEOUT) <= time_in_seconds * 1000000) {
+            LogPrintf("Disconnecting outbound peer %d due to tx request backlog (%d inflight, %d queued)\n", pto->GetId(), tx_download.m_tx_in_flight.size(), tx_download.m_tx_process_time.size());
```

Comment:
This will disconnect both inbound and outbound peers, I think?

Reply:
Yeah, fixed